src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingFileSystemWatcher.cs

```diff
@@ -167,20 +167,26 @@ internal void UpdateState(string directory, ref FileChangeList changes, ref File
         /// </summary>
         public void Dispose()
         {
-            _disposed = true;
-            _timer.Dispose();
-            Dispose(true);
-            GC.SuppressFinalize(this);
+            lock (_timer)
```

Comment:
We clearly need some sort of synchronization mechanism here or else we keep getting intermittent test failures here: https://github.com/dotnet/corefxlab/pull/2603/files#diff-bf9f0af07091216198ca86b115fd2777R213

@JeremyKuhne, @MisinformedDNA, does this change of using a lock during Dispose make sense to you?

Otherwise, we should skip all the Polling tests against https://github.com/dotnet/corefxlab/issues/420#issuecomment-441950347.

Reply:
Never mind. That didn't resolve the issue. Will revert.

Reply:
I don't fully understand why it fails at all. I believe it is only certain platforms. With that and many other issues (like lack of extensibility hooks I needed), I [proposed](https://github.com/dotnet/corefxlab/issues/2390) a base class for PollingFileSystemWatcher that would make the unit testing much easier, but the discussion stalled out.

Reply:
> I believe it is only certain platforms.

I was able to repro the failures on windows (x64, debug) for all the tests that I have marked as skip. It is a (or several) race condition(s) somewhere.

Many of the tests fail in < 100 iterations, but some took 1000s to repro, locally.

```text
Unhandled Exception: System.ObjectDisposedException: Cannot access a disposed object.
```

Reply:
If anyone wants the test projects that I ran to consistently repro the failures (essentially its just separate applications that run a single test in a loop), here it is:
https://github.com/ahsonkhan/corefxlab/tree/PollingTestRepros/repro_test_failures

Run them like any dotnet application.
`SerializableTests\RoundTripAfterStartedTest> dotnet run`

Or you could just add a loop in the test itself and you should be able to see the failure in VS test explorer.